src/components/Table.tsx

```diff
@@ -0,0 +1,592 @@
+    header: "Length",
+    cell: (info: any) => {
+      return info.getValue()
+    },
+  },
+]
+
+const Table = () => {
+  const searchQuery = useAppSelector((state) => state.search.searchQuery)
+
+  const selectedFilters = useAppSelector(
+    (state) => state.search.selectedFilters,
+  )
+
+  const navigate = useNavigate()
+  const dispatch = useAppDispatch()
+
+  const tableContainerRef = React.useRef<HTMLDivElement>(null)
+
+  const [sorting, setSorting] = useState<SortingState>([])
+  const [cursor, setCursor] = useState<string | undefined>("")
+  const [globalFilter, setGlobalFilter] = useState<string>("")
+  const [hasElements, setHasElements] = useState(false)
+  const [totalResults, setTotalResults] = useState("0")
+  const [sort, setSort] = useState({ id: "", type: "" })
+
+  useEffect(() => {
+    const queryParams = new URLSearchParams(window.location.search)
+    const sortValue = queryParams.get("sort") ?? ""
+
+    if (sortValue && sort.id === "") {
+      const decoded = decodeURIComponent(sortValue)
+      const [id, type] = decoded.split(" ")
+
+      setSort({ id, type })
+    }
+  }, [])
+
+  useEffect(() => {
+    const queryParams = new URLSearchParams(window.location.search)
+    const sortValue = queryParams.get("sort") ?? sort.id
+
+    if (sortValue && sort.id !== "") {
+      queryParams.set("sort", `${sort.id} ${sort.type}`)
+    } else {
+      queryParams.delete("sort")
+    }
+
+    updateQueryParam(queryParams)
+  }, [sort])
+
+  useEffect(() => {
+    const queryParams = new URLSearchParams(window.location.search)
+    const urlFilters = queryParams.get("filters") ?? ""
+
+    if (urlFilters && selectedFilters === null) {
+      const decoded = decodeURIComponent(urlFilters)
+      const filters = JSON.parse(decoded)
+
+      dispatch(setFilters(filters))
+    }
+  }, [])
+
+  useEffect(() => {
+    const queryParams = new URLSearchParams(window.location.search)
+
+    if (selectedFilters === null) {
+      queryParams.delete("filters")
+    } else {
+      const encoded = encodeURIComponent(JSON.stringify(selectedFilters))
+
+      queryParams.set("filters", encoded)
+    }
+
+    updateQueryParam(queryParams)
+  }, [selectedFilters])
+
+  const handleSort = (id: string) => {
+    setCursor("")
+
+    if (sort.id !== id) {
+      setSort({ id, type: "asc" })
+    } else if (sort.id === id && sort.type === "desc") {
+      setSort({ id: "", type: "" })
+    } else if (sort.id === id && sort.type === "asc") {
+      setSort({ id, type: "desc" })
+    }
+
+    return
+  }
+
+  useEffect(() => {
+    setCursor("")
+  }, [searchQuery, selectedFilters])
+
+  const updateQueryParam = useCallback(
+    (params: URLSearchParams) => {
+      const search = params.toString()
+
+      navigate({ search })
+    },
+    [navigate],
+  )
+
+  const fetchData = useCallback(
+    async ({ pageParam = "" }) => {
+      const query = globalFilter || "*"
+      const queryParams = new URLSearchParams(window.location.search)
```

Comment:
what is the responsibility of `fetchData` function? To fetch data or to handle URLSearchParams?